### PR TITLE
Feature: release checklist

### PR DESCRIPTION
**Pull request by marcelosalloum**
<details>
  <summary>PR Checklist</summary>

### PR Structure

* [x] This PR has reasonably narrow scope (if not, break it down into smaller PRs).
* [x] This PR avoids mixing refactoring changes with feature changes (split into two PRs
  otherwise).
* [x] This PR's title starts with name of package that is most changed in the PR, ex.
  `paymentservice.stellar`, or `all` or `doc` if the changes are broad or impact many
  packages.

### Thoroughness

* [x] This PR adds tests for the most critical parts of the new functionality or fixes.
</details>

### What

Add a release checklist to be followed before making a new release.

With this addition, we will have a new issue option called `Release a new Version!` when creating a [new issue](https://github.com/stellar/java-stellar-anchor-sdk/issues/new/choose). The idea is that we create a ticket for new releases and follow the checklist to make sure the release process is being followed thoroughly.

### Why

As part of the new gitflow process outlined in #487.

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/release_a_new_version.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+- [ ] Update the code to use this version number.
+- [ ] Update the [CHANGELOG.md] file with the new version number and release notes.
+- [ ] Run tests and linting. **Not only CI/CD tests, but also manual tests to make sure the release is up and running, and that it's stable!**
+- [ ] Make all changes necessary to make sure the release is ready to be published. If new issues are found on manual tests, improve the CI so these issues can be automatically detected next time.
+- [ ] DO NOT RELEASE before holidays or weekends! Mondays and Tuesdays are preferred.
+- [ ] When the team is confident the release is stable, you'll need to create two pull requests:
+  - [ ] `release/0.1.0 -> main`: this should require two approvals.
+  - [ ] `release/0.1.0 -> develop`: ideally, this should be merged after the `main` branch is merged.
+- [ ] Create a new release on GitHub with the name `0.1.0` and the changes from the [CHANGELOG.md] file.
+  - The release should trigger the release process on GitHub that will:
```
Comment (erika-sdf) [inline .github/ISSUE_TEMPLATE/release_a_new_version.md:26]: Do these things currently happen? If not, should we make a note that this has yet to be automated?

File: .github/ISSUE_TEMPLATE/release_a_new_version.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+> Attention: the examples below use the version `0.1.0` but you should update them to use the version you're releasing.
+
+- [ ] Decide on a version number based on the current version number and the common rules defined in [Semantic Versioning](https://semver.org). E.g. `0.1.0`.
+- [ ] Update this ticket name to reflect the new version number, following the pattern "Release `0.1.0`".
+- [ ] Cut a branch for the new release out of the `develop` branch, following the gitflow naming pattern `release/0.1.0`.
+- [ ] Update the code to use this version number.
+- [ ] Update the [CHANGELOG.md] file with the new version number and release notes.
+- [ ] Run tests and linting. **Not only CI/CD tests, but also manual tests to make sure the release is up and running, and that it's stable!**
+- [ ] Make all changes necessary to make sure the release is ready to be published. If new issues are found on manual tests, improve the CI so these issues can be automatically detected next time.
```
Comment (erika-sdf) [inline .github/ISSUE_TEMPLATE/release_a_new_version.md:20]: We should definitely ensure that tests pass, but I don't know if we want to block releases on adding tests to the CI. What if we suggest filing an issue here instead?

**Review by erika-sdf — APPROVED** [review]
Awesome! Thanks for doing this!

File: .github/ISSUE_TEMPLATE/release_a_new_version.md (hunk 2 above)
Comment (marcelosalloum) [reply .github/ISSUE_TEMPLATE/release_a_new_version.md:20]: Good point, I'll update it to create new issues to automate these tests.

File: .github/ISSUE_TEMPLATE/release_a_new_version.md (hunk 1 above)
Comment (marcelosalloum) [reply .github/ISSUE_TEMPLATE/release_a_new_version.md:26]: Good point, I'll add a WIP tag there.

**Pull request merged**